pkg/gadgets/socket-collector/tracer/tracer.go

```diff
@@ -183,15 +173,18 @@ func RunCollector(pid uint32, filter *gadgetv1alpha1.ContainerFilter) (string, e
 				return "", err
 			}
 
-			sockets = append(sockets, Socket{
-				Protocol:            proto,
-				LocalAddress:        parseIPv4(src),
-				LocalPort:           srcp,
-				RemoteAddress:       parseIPv4(dest),
-				RemotePort:          destp,
-				Status:              status,
-				KubernetesNamespace: filter.Namespace,
-				KubernetesPod:       filter.Podname,
+			sockets = append(sockets, socketcollectortypes.Event{
+				Event: eventtypes.Event{
+					Node:      node,
+					Namespace: namespace,
+					Pod:       podname,
+				},
```

Comment:
Note: it's fine for now but once you start supporting filtering by namespace, labels and so on this will have to be changed as the namespace, node and podname could be different for each event.